pkg/controller/model/grafanaService.go

```diff
@@ -107,7 +107,7 @@ func GrafanaService(cr *v1alpha1.Grafana) *v1.Service {
 			Selector: map[string]string{
 				"app": GrafanaPodLabel,
 			},
-			ClusterIP: "",
+			ClusterIP: cr.Spec.Service.ClusterIP,
```

Comment:
This causes a segfault when `Service` (an optional field in the CR) is nil.

Reply:
Something like this should do:

```
func getClusterIP(cr *v1alpha1.Grafana) string {
	if cr.Spec.Service == nil {
		return ""
	}
	return cr.Spec.Service.ClusterIP
}
```

Reply:
you are right, PR updated. :)